atlp-rwanda/eagles-ec-fe
# #187419150 : ft notifications(right user should receive notification)

## What does this PR do?
This PR implements real-time notifications whenever product lifecycle changes.

## Task completed
- Integrated Socket.IO for real-time communication
- Ensure notifications are sent to the correct user (seller).
- Updated frontend to display notifications
- Added a notification icon to show the count of unread notifications
- Implemented a detailed view for all notifications



## How can this be tested?
1. Register account as seller ,verify your account or login if u already has an account [here](https://eagles-ec-fe-git-ft-notification-187419150-eagles-ec.vercel.app/)
1. Perform different operations on a product (add, delete or update).
3. Observe the notification icon for new notifications.
4. Click on the notification icon to view all notifications.

## Pivot tracker ID -
#187419150

## Screenshots



<img width="1440" alt="Screenshot 2024-07-10 at 17 01 19" src="https://github.com/atlp-rwanda/eagles-ec-fe/assets/105858896/692f0a81-4f3c-49b3-bb61-11176a961d18">


<img width="1440" alt="Screenshot 2024-07-10 at 16 55 33" src="https://github.com/atlp-rwanda/eagles-ec-fe/assets/105858896/5f52ace8-784f-4091-a2f4-bb476dc6c5bd">

<img width="1440" alt="Screenshot 2024-07-10 at 16 54 34" src="https://github.com/atlp-rwanda/eagles-ec-fe/assets/105858896/e9176db9-02d7-437f-bb35-9681fb3bf8bb">
